R/final_size.R

```diff
@@ -47,12 +47,22 @@ final_size <- function(r0 = 2, contact_matrix, demography_vector,
   # Check inputs
   if (is.null(prop_suscep)) {
     prop_suscep <- rep(1, length(demography_vector))
-  } # Assume fully susceptible if no entry
+  } else {
+    checkmate::assert_numeric(
+      prop_suscep,
+      lower = 0.0, upper = 1.0, finite = TRUE
+    )
+    if (length(prop_suscep) == 1L) {
+      prop_suscep <- rep(prop_suscep, length(demography_vector))
```

Comment:
Another, simpler(?) way to deal with #31 is to allow `prop_suscep` to be of length 1 in L71-74. It will then be automatically recycled.

I like it better because it doesn't burden the function with more default settings and recycling elements of length 1 is very idiomatic in R anyways so I don't think it hurts readability. The mere fact that we're considering adding this behaviour is a proof of that.

Open to discussion on this though.

Reply:
I've updated the asserts around `prop_suscep` to suit.

>     * no bleeding of infections from one age group to the other if the contact rate is set to 0

Is referenced in #33

>     * higher values of R0 result in higher finalsizes (all else being constant)

Already tested

>     * finalsize cannot exceed prop_suscep

This is a bit tricky, because I'm not sure that `prop_suscep` is actually the proportion susceptible - maybe the epidemiologists could weigh in on how this test should be structured?